www/js/metrics/footprintHelper.ts

```diff
@@ -0,0 +1,104 @@
+import { displayErrorMsg, logDebug, logWarn } from '../plugin/logger';
+import { getCustomFootprint } from './customMetricsHelper';
+
+//variables for the highest footprint in the set and if using custom
+let highestFootprint = 0;
+
+/**
+ * @function converts meters to kilometers
+ * @param {number} v value in meters to be converted
+ * @returns {number} converted value in km
+ */
+const mtokm = function (v) {
+  return v / 1000;
+};
+
+/**
+ * @function clears the stored highest footprint
+ */
+export const clearHighestFootprint = function () {
+  //need to clear for testing
+  highestFootprint = undefined;
+};
+
+/**
+ * @function gets the footprint
+ * currently will only be custom, as all labels are "custom"
+ * fallback is json/label-options.json.sample, with MET and kgCO2 defined
+ * @returns the footprint or undefined
+ */
+const getFootprint = function () {
+  let footprint = getCustomFootprint();
+  if (footprint) {
+    return footprint;
+  } else {
+    displayErrorMsg('failed to use custom labels', 'Error in Footprint Calculatons');
+    return undefined;
+  }
+};
+
+/**
+ * @function calculates footprint for given metrics
+ * @param {Array} userMetrics string mode + number distance in meters pairs
+ * ex: const custom_metrics = [ { key: 'walk', values: 3000 }, { key: 'bike', values: 6500 }, ];
+ * @param {number} defaultIfMissing optional, carbon intensity if mode not in footprint
+ * @returns {number} the sum of carbon emissions for userMetrics given
+ */
+export const getFootprintForMetrics = function (userMetrics, defaultIfMissing = 0) {
+  var footprint = getFootprint();
+  logDebug('getting footprint for ' + userMetrics + ' with ' + footprint);
+  var result = 0;
+  for (var i in userMetrics) {
+    var mode = userMetrics[i].key;
+    if (mode == 'ON_FOOT') {
+      mode = 'WALKING';
+    }
+
+    if (mode in footprint) {
+      result += footprint[mode] * mtokm(userMetrics[i].values);
+    } else if (mode == 'IN_VEHICLE') {
+      result +=
+        ((footprint['CAR'] +
+          footprint['BUS'] +
+          footprint['LIGHT_RAIL'] +
+          footprint['TRAIN'] +
+          footprint['TRAM'] +
+          footprint['SUBWAY']) /
+          6) *
```

Comment:
Ah, I had misread that
Maybe there's something we can do with the formatting to prevent confusion

Reply:
we can store the sum and then divide?
<img width="496" alt="Screenshot 2023-11-17 at 3 09 30 PM" src="https://github.com/e-mission/e-mission-phone/assets/54848919/f047528d-d781-4049-b2fd-1a314a3a7f06">

Reply:
I went ahead and pushed the change, with the way prettier chose to format the two, this is a line shorter! [e626f6e](https://github.com/e-mission/e-mission-phone/pull/1098/commits/e626f6e36c5b5cdadfb16968f03204cf024b2066)